src/makeCalculator.js

```diff
@@ -4,7 +4,53 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  return {
+    add(num) {
+      result += num;
+
+      return this;
+    },
+
+    subtract(num) {
+      result -= num;
+
+      return this;
+    },
+
+    multiply(num) {
+      result *= num;
+
+      return this;
+    },
+
+    divide(num) {
+      if (num !== 0) {
+        result /= num;
+      } else {
+        throw new Error('Division by zero is not allowed');
+      }
+
+      return this;
+    },
+
+    operate(callback, number) {
+      callback.call(this, number);
+
+      return this;
```

Comment:
The `operate` method correctly takes a callback and a number, applies the callback, and allows for chaining by returning `this`. However, ensure that the callback is a function before calling it to avoid runtime errors.